rfcs/Subscriptions.md

```diff
@@ -0,0 +1,94 @@
+RFC: GraphQL Subscriptions
+-------
+
+*Working Draft - February 2017*
+
+**Introduction**
+
+Modern applications increasingly include scenarios where users receive immediate feedback when something interesting happens. When these applications outgrow simple polling solutions, they turn to stateful, bi-directional solutions that push data from server to client. There are many ways to extend GraphQL to support push-based solutions. By standardizing these concepts and solutions, we can keep the community cohesive, and ensure sustainable future improvements.
+
+**Background**
+
+Using the standard request/response GraphQL model, apps can support realtime scenarios by repeatedly issuing the same GraphQL query. However, this solution is inefficient and large scale apps may prefer push-based solutions. Compared to polling, push-based solutions trade off statelessness and simplicity for efficiency while raising new questions and challenges. For example: how is an “interesting event” detected and propagated? How does the GraphQL language express these semantics? How does this affect the client-side GraphQL API?
+
+There are multiple valid answers to these questions, several of which have been proposed by the open-source community already [1][2][3]. At Facebook, we have been using “GraphQL Subscriptions” [4] at scale since 2015. Following our open-source philosophy, we now believe this system is useful and generalizable. We would like to contribute our ideas to the conversation.
+
+One of GraphQL's superpowers is allowing clients to specify exactly the data they need, sidestepping a problem we call “over/under-fetching”. With push-based solutions, we have a similar problem: “over/under pushing.” Under-pushing means we didn't send enough data, and the client now needs to make a follow-up network request. Over-pushing means we sent the client too much data; in the best case this data is irrelevant and doesn't waste anybody's mobile data plan, in the worst case, this data contains sensitive information that the client should not see. Just as request/response GraphQL addresses over and under-fetching, we believe GraphQL Subscriptions addresses over/under-pushing.
+
+**Possible Solutions**
+
+We broadly categorize realtime API solutions into three types:
```

Comment:
Not sure if is a typo, but there are *six* types defined below (instead of the *three* mentioned here).